WMI::IsBitlockedDrive: check ConversionStatus rather than ProtectionStatus

ramcq opened the pull request:
A drive can be encrypted (partially or fully) without the key being sealed by the TPM which makes it protected. Conversion status tells us whether the bytes on disk are in clear or not.

https://phabricator.endlessm.com/T21417

ramcq commented:
Untested bilge. Let's see what Beatriz finds.

wjt commented on src/endless/WMI.cpp:
```diff
@@ -102,7 +102,7 @@ BOOL WMI::IsBitlockedDrive(const CString & drive)
     CString bitlockerQuery;
 
-    bitlockerQuery.Format(L"Select * from Win32_EncryptableVolume where ProtectionStatus != 0 and DriveLetter = '%ls'", drive);
+    bitlockerQuery.Format(L"Select * from Win32_EncryptableVolume where ConversionStatus != 0 and DriveLetter = '%ls'", drive);
```
Well spotted. Can we add a link to https://msdn.microsoft.com/en-us/library/windows/desktop/aa376483(v=vs.85).aspx in this function? And yes, let's wait and see what Beatriz comes back with.

wjt committed:
WMI: check ConversionStatus as well as ProtectionStatus

Previously, we only checked ProtectionStatus, but as explained in the
code it's possible for the disk to be encrypted even if ProtectionStatus
== 0.

Rather than including the new condition in the query, let's always fetch
the record for the drive letter in question, and explicitly print the
properties we're checking. It would be nice to print all its properties
but today's not the day I write a generic function to print a VARIANT's
type and value. (Although perhaps _variant_t's implicit conversion to
string would do it?)

https://phabricator.endlessm.com/T21417

wjt commented:
@ramcq how's that?

ramcq commented:
Looks legit.
